lib/wasi/src/os/task/control_plane.rs

```diff
@@ -0,0 +1,206 @@
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc, RwLock,
+    },
+};
+
+use crate::{WasiProcess, WasiProcessId};
+
+#[derive(Debug, Clone)]
+pub struct WasiControlPlane {
+    state: Arc<State>,
+}
```

Comment:
Can we name this thing somewhat different? In my mind a control plane is related to higher-infra stuff.
Thoughts? What would be a better name?

Reply:
Happy to hear suggestions.

This doesn't currently hold any actual logic, just the data, so `ProcessTree` ? 

But we should probably move some of the process and thread spawning logic in here, so `ProcessManager`?

Reply:
Didn't get any suggestions - name is fine for now.